Sources/Core/Extensions/Other/XCTestCase+PovioKit.swift

```diff
@@ -12,9 +12,38 @@ public extension XCTestCase {
   func XCTAssertEqualURLRequest(_ lhs: URLRequest?, _ rhs: URLRequest?, file: StaticString = #filePath, line: UInt = #line) {
     XCTAssertEqual(lhs, rhs, file: file, line: line)
     guard let lhs = lhs, let rhs = rhs else { return }
-    XCTAssertEqual(lhs.url, rhs.url, "url", file: file, line: line)
-    XCTAssertEqual(lhs.httpMethod, rhs.httpMethod, "httpMethod", file: file, line: line)
-    XCTAssertEqual(lhs.httpBody ?? Data(), rhs.httpBody ?? Data(), "httpBody", file: file, line: line)
-    XCTAssertEqual(lhs.allHTTPHeaderFields ?? [:], rhs.allHTTPHeaderFields ?? [:], "allHTTPHeaderFields", file: file, line: line)
+    XCTAssertEqual(lhs.url, rhs.url, "Url doesn't equal!", file: file, line: line)
+    XCTAssertEqual(lhs.httpMethod, rhs.httpMethod, "HttpMethod doesn't equal!", file: file, line: line)
+    XCTAssertEqual(lhs.httpBody ?? Data(), rhs.httpBody ?? Data(), "HttpBody doesn't equal!", file: file, line: line)
+    XCTAssertEqual(lhs.allHTTPHeaderFields ?? [:], rhs.allHTTPHeaderFields ?? [:], "AllHTTPHeaderFields doesn't equal!", file: file, line: line)
+  }
+  
+  func XCTAssertEqualImage(_ lhs: UIImage?, _ rhs: UIImage?, file: StaticString = #filePath, line: UInt = #line) {
+    guard let image1 = lhs, let image2 = rhs,
+          let imageData1 = image1.pngData(),
+          let imageData2 = image2.pngData() else { return }
+    XCTAssertTrue(imageData1.elementsEqual(imageData2), "Images doesn't equal!", file: file, line: line)
+  }
+  
+  func XCTAssertNotEqualImage(_ lhs: UIImage?, _ rhs: UIImage?, file: StaticString = #filePath, line: UInt = #line) {
+    guard let image1 = lhs, let image2 = rhs,
+          let imageData1 = image1.pngData(),
+          let imageData2 = image2.pngData() else { return }
+    XCTAssertFalse(imageData1.elementsEqual(imageData2), "Images do equal!", file: file, line: line)
+  }
+  
+  func XCTAssertEqualFont(lhs: UIFont?, rhs: UIFont?, file: StaticString = #filePath, line: UInt = #line) {
+    XCTAssertEqual(lhs?.pointSize, rhs?.pointSize, "PointSize doesn't equal!", file: file, line: line)
+    XCTAssertEqual(lhs?.fontName, rhs?.fontName, "FontName doesn't equal!", file: file, line: line)
+    XCTAssertEqual(lhs?.familyName, rhs?.familyName, "FamilyName doesn't equal!", file: file, line: line)
+  }
+  
+  func XCTAssertNotEqualFont(lhs: UIFont?, rhs: UIFont?, file: StaticString = #filePath, line: UInt = #line) {
+    guard let lhs = lhs, let rhs = rhs else { return }
+    let pointSizeEquals = lhs.pointSize == rhs.pointSize
+    let fontNameEquals = lhs.fontName == rhs.fontName
+    let familyNameEquals = lhs.familyName == rhs.familyName
+    guard pointSizeEquals, fontNameEquals, familyNameEquals else { return }
+    XCTFail("Fonts do equal!", file: file, line: line)
```

Comment:
I would do asserts for each property that we are testing, otherwise we lose the ability to distinguish what properties are equal in this particular test.

Reply:
Not really. If you have the same font, only pointSize differs. Asserting to no equal pointSize will succeed, but the other two asserts will fail, therefore test will fail.

```swift
let lhs: UIFont? = UIFont.systemFont(ofSize: 12)
let rhs: UFont? = UIFont.systemFont(ofSize: 13)
XCTAssertNotEqual(lhs?.pointSize, rhs?.pointSize, "PointSize equals!", file: file, line: line) - succeeds
XCTAssertNotEqual(lhs?.fontName, rhs?.fontName, "FontName equals!", file: file, line: line) - fails
XCTAssertNotEqual(lhs?.familyName, rhs?.familyName, "FamilyName equals!", file: file, line: line) - fails
```

Reply:
Ahh, so the idea is that if only one of those properties is different then we mark the fonts as different.